image_picker/file_windows.go

```diff
@@ -0,0 +1,101 @@
+package image_picker
+
+import (
+	"strings"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	comdlg32         = syscall.NewLazyDLL("comdlg32.dll")
+	getOpenFileNameW = comdlg32.NewProc("GetOpenFileNameW")
+)
+
+const (
+	maxPath          = 260
+	ofnExplorer      = 0x00080000
+	ofnFileMustExist = 0x00001000
+	ofnHideReadOnly  = 0x00000004
+)
+
+type openfilenameW struct {
```

Comment:
Are all the fields require for the plugin to work?
Only 6 of the 23 fields are used.

Reply:
it's require for the `uint32(unsafe.Sizeof(ofn))`.. not that pretty, but we don't have the choice

Reply:
I had tried. but dialog will not show. 
I think this c++ strunct can't modify 
https://docs.microsoft.com/en-us/windows/win32/api/commdlg/ns-commdlg-openfilenamew

typedef struct tagOFNW {
  DWORD         lStructSize;
  HWND          hwndOwner;
  HINSTANCE     hInstance;
  LPCWSTR       lpstrFilter;
  LPWSTR        lpstrCustomFilter;
  DWORD         nMaxCustFilter;
  DWORD         nFilterIndex;
  LPWSTR        lpstrFile;
  DWORD         nMaxFile;
  LPWSTR        lpstrFileTitle;
  DWORD         nMaxFileTitle;
  LPCWSTR       lpstrInitialDir;
  LPCWSTR       lpstrTitle;
  DWORD         Flags;
  WORD          nFileOffset;
  WORD          nFileExtension;
  LPCWSTR       lpstrDefExt;
  LPARAM        lCustData;
  LPOFNHOOKPROC lpfnHook;
  LPCWSTR       lpTemplateName;
  LPEDITMENU    lpEditInfo;
  LPCSTR        lpstrPrompt;
  void          *pvReserved;
  DWORD         dwReserved;
  DWORD         FlagsEx;
} OPENFILENAMEW, *LPOPENFILENAMEW;

so I rollback again.
should I put comment after each field?
Thanks

Reply:
ok the order has the be the same.